server/asset/btc/rpcclient.go

```diff
@@ -0,0 +1,142 @@
+// This code is available on the terms of the project LICENSE.md file,
+// also available online at https://blueoakcouncil.org/license/1.0.0.
+
+package btc
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/btcsuite/btcd/btcjson"
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcutil"
+)
+
+const (
+	methodGetBestBlockHash  = "getbestblockhash"
+	methodGetBlockchainInfo = "getblockchaininfo"
+	methodEstimateSmartFee  = "estimatesmartfee"
+	methodGetTxOut          = "gettxout"
+	methodGetRawTransaction = "getrawtransaction"
+	methodGetBlock          = "getblock"
+)
+
+// RawRequester defines dcrd's rpcclient RawRequest func where all RPC
+// requests sent through. For testing, it can be satisfied by a stub.
+type RawRequester interface {
+	RawRequest(context.Context, string, []json.RawMessage) (json.RawMessage, error)
+	Shutdown()
+	WaitForShutdown()
+}
+
+// RPCClient is a bitcoind wallet RPC client that uses rpcclient.Client's
+// RawRequest for wallet-related calls.
+type RPCClient struct {
+	ctx       context.Context
+	requester RawRequester
+}
+
+func (rc *RPCClient) callHashGetter(method string, args anylist) (*chainhash.Hash, error) {
+	var txid string
+	err := rc.call(method, args, &txid)
+	if err != nil {
+		return nil, err
+	}
+	return chainhash.NewHashFromStr(txid)
+}
+
+// GetBestBlockHash returns the hash of the best block in the longest block
+// chain.
+func (rc *RPCClient) GetBestBlockHash() (*chainhash.Hash, error) {
+	return rc.callHashGetter(methodGetBestBlockHash, nil)
+}
+
+// GetBlockchainInfoResult models the data returned from the getblockchaininfo
+// command.
+type GetBlockchainInfoResult struct {
+	Blocks               int64  `json:"blocks"`
+	Headers              int64  `json:"headers"`
+	BestBlockHash        string `json:"bestblockhash"`
+	InitialBlockDownload bool   `json:"initialblockdownload"`
+}
+
+// GetBlockChainInfo returns information related to the processing state of
+// various chain-specific details.
+func (rc *RPCClient) GetBlockChainInfo() (*GetBlockchainInfoResult, error) {
+	chainInfo := new(GetBlockchainInfoResult)
+	err := rc.call(methodGetBlockchainInfo, nil, chainInfo)
+	if err != nil {
+		return nil, err
+	}
+	return chainInfo, nil
+}
+
+// EstimateSmartFee requests the server to estimate a fee level based on the
+// given parameters.
```

Comment:
The general statement `based on the given parameters` could be appended to the end of almost any docstring. It doesn't add anything, really. If parameters are important enough to mention, you should be specific about the parameters you think are worth mentioning, e.g. `based on a target confirmation level`. If there are none worth mentioning specifically, then you don't need to mention parameters at all. It's okay to let a function signature with well-named parameters speak for itself.

Reply:
Agreed.